client-course-schedulizer/src/utilities/helpers/writeCSV.ts

```diff
@@ -1,41 +1,115 @@
-import { Schedule } from "utilities/interfaces";
+import moment from "moment";
+import { Day, Schedule, Term } from "utilities/interfaces";
 
 export const scheduleToCSVString = (schedule: Schedule): string => {
+  const numericReg = RegExp("[0-9]");
   let csvStr =
-    "name,prefixes,number,section,studentHours,facultyHours,startTime,duration,location,roomCapacity,year,term,semesterLength,days,globalMax,localMax,anticipatedSize,instructors,comments";
+    "Department,Term,TermStart,AcademicYear,SectionName,SubjectCode,CourseNum,SectionCode,CourseLevelCode,MinimumCredits,FacultyLoad,Used,Day10Used,LocalMax,Global Max,RoomCapacity,BuildingAndRoom,MeetingDays,MeetingTime,SectionStartDate,SectionEndDate,SemesterLength,Building,RoomNumber,MeetingStart,MeetingStartInternal,MeetingDurationMinutes,MeetingEnd,MeetingEndInternal,Monday,Tuesday,Wednesday,Thursday,Friday,ShortTitle,Faculty,SectionStatus,InstructionalMethod,Comments";
   schedule.courses.forEach((course) => {
     course.sections.forEach((section) => {
       // Iterate through meetings to construct relevant strings
-      let startTimeStr = "";
-      let durationStr = "";
-      let locationStr = "";
+      let startMoment;
+      let meetingTimeStr = "";
+      let meetingStartStr = "";
+      let meetingStartInternalStr = "";
+      let meetingEndStr = "";
+      let meetingEndInternalStr = "";
+      let meetingDurationMinutesStr = "";
+      let buildingStr = "";
+      let roomNumberStr = "";
+      let buildingAndRoomStr = "";
       let roomCapacityStr = "";
       let daysStr = "";
+      let monStr = "";
+      let tuesStr = "";
+      let wedStr = "";
+      let thursStr = "";
+      let friStr = "";
       section.meetings.forEach((meeting) => {
-        startTimeStr += `${meeting.startTime}\n`;
-        durationStr += `${meeting.duration}\n`;
-        locationStr += meeting.location.roomNumber
+        startMoment = moment(meeting.startTime, "h:mm A");
+        if (startMoment.isValid()) {
+          meetingTimeStr += `${startMoment.format("h:mmA")} - ${startMoment
+            .add(meeting.duration, "minutes")
+            .format("h:mm A")}\n`;
+          meetingStartStr += `${startMoment.format("h:mm A")}\n`;
+          meetingStartInternalStr += `${startMoment.format("H:mm:ss")}\n`;
```

Comment:
When you add to a moment, it reassigns the moment to the new value even if you don't explicitly reassign it. So, in this case, adding to `startMoment` with `.add()`when computing the end time results in `meetingStartStr` and `meetingStartInternalStr` equalling the end time rather than the start time. You'll have to make a copy of `startMoment` when you add to it.

Reply:
Good catch: 1e28f58